lib/std/testing.zig

```diff
@@ -466,8 +466,13 @@ test {
 pub fn refAllDecls(comptime T: type) void {
     if (!builtin.is_test) return;
     inline for (comptime std.meta.declarations(T)) |decl| {
-        if (decl.is_pub and @typeInfo(@TypeOf(@field(T, decl.name))) == .Struct)
-            _ = @hasDecl(@field(T, decl.name), "foo");
+        if (decl.is_pub) {
+            const decl_value = @field(T, decl.name);
+            const type_of_decl = @TypeOf(decl_value);
+            if (type_of_decl == type and @typeInfo(type_of_decl) == .Struct) {
```

Comment:
This is always false but it seems that just doing `@field(T, decl.name)` is enough to force the declaration to be analyzed.